src/profile/hooks/useProfileLoadTracker.ts

```diff
@@ -0,0 +1,119 @@
+import { useCallback, useContext, useEffect, useRef } from 'react';
+
+import { ProfileContext, ProfileContextData } from '../context/ProfileContext';
+import parseGraphQLError from '../helpers/parseGraphQLError';
+import { QueryResult } from './useProfileQuery';
+
+export const STATUS_FAILED = -1;
+export const STATUS_WAITING = 0;
+export const STATUS_SHOULD_LOAD = 1;
+export const STATUS_LOADING = 2;
+export const STATUS_SUCCESS = 3;
+
+export type Status =
+  | typeof STATUS_WAITING
+  | typeof STATUS_LOADING
+  | typeof STATUS_FAILED
+  | typeof STATUS_SUCCESS
+  | typeof STATUS_SHOULD_LOAD;
+
+export type useProfileLoaderHookReturnType = {
+  isProfileLoadComplete: () => boolean;
+  hasExistingProfile: () => boolean;
+  didProfileLoadFail: () => boolean;
+  reloadProfile: () => Promise<QueryResult>;
+  getStatus: () => Status;
+  getProfile: () => ReturnType<ProfileContextData['getProfile']>;
```

Comment:
The `getStatus` and `getProfile` functions don't seem to be used in any production code, so can they be removed? If `getStatus` is not needed, perhaps the `STATUS_*` consts and `Status` type don't have to exported either.